### ISSUE TITLE
occasional connect ETIMEDOUT 54.93.180.229:443 while sending audio

### ISSUE THREAD

**Issue by talhabalaj** [adiwajshing/Baileys#136]
Hello!

I am having this occasional issue: sending audio throws a connection timeout. I am using GCP, you chance that I'm having a connection issue is not possible. I have no idea why it occurs. As I mentioned this is occasional - sometimes it works other times throws this. 

```
FetchError: request to https://mmh.whatsapp.net/mms/audio/QTpbNH0fPM1nUbk4K80UENpRHfIqq9-mBrcpiQaTF7Y?auth=AWTTcBxGWPBdpyELnWC2saWnhaPayuLicCaE643WZO8LmzD4-qHzHOALcA0&token=QTpbNH0fPM1nUbk4K80UENpRHfIqq9-mBrcpiQaTF7Y failed, reason: connect ETIMEDOUT 54.93.180.229:443
```

Thank you!

**Comment by adiwajshing** [comment]
This actually seems like a connectivity issue on GCP -- it's possible that this your assigned IP may be blocked by WA or something

**Comment by talhabalaj** [comment]
Thank you, It's probably WA blocking my IP Address. Automating WA isn't an easy task. They must have some kind of mechanism to detect automation. Hopefully, you can figure this out!

**Comment by adiwajshing** [comment]
I've updated the library to try all possible hosts for a media upload. Please try out the latest commit & lemme know if that works.

**Comment by talhabalaj** [comment]
Nice. Let me test it out.

**Comment by talhabalaj** [comment]
First upload worked, following uploads threw `Media Upload failed on all hosts`

**Comment by adiwajshing** [comment]
Can you try it on your local machine?

**Comment by talhabalaj** [comment]
It's now working on GCP aswell.

**Comment by talhabalaj** [comment]
It's still randomly working and randomly not working :(

**Comment by adiwajshing** [comment]
Did you try it out on your local machine?

**Comment by talhabalaj** [comment]
Yeah, I tried it on my local machine. It stops working after some successful sends.

**Comment by adiwajshing** [comment]
1. How many sends on an average till it starts giving an error?
2. What platform are you running the library on?
3. Can you write a test to reproduce the issue? Maybe something like the following:
``` ts
it('should upload media successfully', async () => {
        const content = await fs.readFile('./the-audio-that-does-not-upload.mp3')
        // run 10 uploads serially
        for (let i = 0; i < 10;i++) {
            await conn.prepareMessageContent (content, MessageType.audio, { filename: 'audio.mp3', mimetype: Mimetype.mp4Audio })
        }
        // run 10 concurrent uploads
        await Promise.all ([...Array(10)]).map (c => (
             conn.prepareMessageContent (content, MessageType.audio, { filename: 'audio.mp3', mimetype: Mimetype.mp4Audio })
        ))
})
```

**Comment by talhabalaj** [comment]
1. It is totally random (after 10 maybe?), Now I haven't got it for a while on my local machine. 
On GCP, I also reserved a new static IP address. I'll update you if it occurs again.
2. Linux 
3. I'm doing the same thing as in the test.

**Comment by talhabalaj** [comment]
Update: It stops working on GCP after 1 or 2 sends. There must be something wrong with the location of my VM Instance that hosts both the android VM & Baileys. 

After updating to the latest commit on my local machine I haven't faced the issue till now so I'm closing the issue.

**Comment by talhabalaj** [comment]
**Update**: WhatsApp blocks a server If it uploads a lot. I shared my bot with my friends and they requested too many songs which caused the media uploading functionality to not work. Rate-Limiting also kicks in after 30-40 requests in a short amount of time. For this to work, Baileys need to bypass all the server stuff.

For the time being, I used Heroku to host the app, now whenever any rate-limiting is applied I just restart the dyno and Heroku allocates a new IP address to the app that works fine.
